aloha-core/src/test/scala/com/eharmony/aloha/models/multilabel/MultilabelModelTest.scala

```diff
@@ -47,124 +63,336 @@ class MultilabelModelTest extends ModelSerializationTestHelper {
 
-    fail()
+    // In this scenario no problems are found but an
+    // Option(GenAggFuncAccessorProblems) is returned instead of None
+    val problemsNoLabelsExpected = Option(GenAggFuncAccessorProblems(Seq(), Seq()))
+    assertEquals(problemsNoLabelsExpected, labelsAndInfoNoLabels.problems)
   }
 
-  @Test def testLabelsForPredictionContainsProblemsWhenLabelsIsEmpty(): Unit = {
-    // Test this:
-    //    val problems =
-    //      if (labelsShouldPredict.nonEmpty) None
-    //      else Option(labelsOfInterest.accessorOutputProblems(a))
+  @Test def testLabelsForPredictionContainsProblemsWhenLabelsIsNotPresent(): Unit = {
+    val labelsExtractor =
+      (label: Map[Label, Label]) => label.get("labels") match {
+        case ls: sci.IndexedSeq[_] => Option(ls.asInstanceOf[sci.IndexedSeq[Label]])
+        case _ => None
+      }
+
+    val descriptor = "label is missing"
+    val labelsOfInterest =
+      GenFunc.f1(GeneratedAccessor(descriptor, labelsExtractor, None))(
+        "", _ getOrElse sci.IndexedSeq.empty[Label]
+      )
+
+    val labelsAndInfoNoLabelsGen1 = labelsForPrediction(
+      example          = Map[Label, Label](),
+      labelsOfInterest = labelsOfInterest,
+      labelToInd       = labelsInTrainingSetToIndex)
+    val problemsNoLabelsExpectedGen1 =
+      Option(GenAggFuncAccessorProblems(Seq(descriptor), Seq()))
+    assertEquals(problemsNoLabelsExpectedGen1, labelsAndInfoNoLabelsGen1.problems)
+  }
 
-    fail()
+  @Test def testLabelsForPredictionContainsProblemsWhenLabelAccessorThrows(): Unit = {
+    val descriptor = "label accessor that throws"
+    val labelsOfInterest =
+      GenFunc.f1(GeneratedAccessor(descriptor,
+        ( _ => throw new Exception()) : Map[String, String] => Option[sci.IndexedSeq[String]], None))(
+        "", _ getOrElse sci.IndexedSeq.empty[Label]
+      )
+    val labelsOfInterestWrapped = EnrichedErrorGenAggFunc(labelsOfInterest)
+
+    val problemsNoLabelsExpected = Option(GenAggFuncAccessorProblems(Seq(), Seq(descriptor)))
+    Try(
+    labelsForPrediction(
+      Map[String, String](),
+      labelsOfInterestWrapped,
+      labelsInTrainingSetToIndex)
+    ).failed.get match {
+      case SemanticsUdfException(_, _, _, accessorsInErr, _, _) =>
+        assertEquals(accessorsInErr, problemsNoLabelsExpected.get.errors)
+    }
   }
 
   @Test def testLabelsForPredictionProvidesLabelsThatCantBePredicted(): Unit = {
-    // Test this:
-    //    val noPrediction =
-    //      if (unsorted.size == labelsShouldPredict.size) Seq.empty
-    //      else labelsShouldPredict.filterNot(labelToInd.contains)
-
-    fail()
+    val labelNotInTrainingSet = "d"
+    val labelsAndInfo = labelsForPrediction(Map("label" -> labelNotInTrainingSet),
+      labelsOfInterestExtractor, labelsInTrainingSetToIndex)
+    assertEquals(Seq(labelNotInTrainingSet), labelsAndInfo.labelsNotInTrainingSet)
   }
 
   @Test def testLabelsForPredictionReturnsLabelsSortedByIndex(): Unit = {
```

Comment:
I would put a comment here that this test would likely be done better in a property-based testing framework like ScalaCheck.